app/src/main/java/io/mochahub/powermeter/workouts/WorkoutFragment.kt

```diff
@@ -41,12 +47,116 @@ class WorkoutFragment : Fragment() {
 
         // TODO: Show a dialog fragment with editor where we can CRUD new session
         addSessionButton.setOnClickListener {
-            viewModel.addWorkout()
+            viewModel.addWorkoutSession()
         }
+
+        enableSwipeToDelete()
     }
 
     // TODO: Show a dialog fragment with editor where we can CRUD this session
     private fun onWorkoutSessionClicked(workoutSession: WorkoutSession) {
         Toast.makeText(requireContext(), "Clicked: ${workoutSession.date}", Toast.LENGTH_LONG).show()
     }
+
+    private fun enableSwipeToDelete() {
+        val simpleItemTouchCallback = object : ItemTouchHelper.SimpleCallback(0, ItemTouchHelper.LEFT or ItemTouchHelper.RIGHT) {
```

Comment:
Can you move this into a reusable abstract class so we can use it in the other screens, and override `onSwiped` in each screen :)

https://github.com/power-meter/android-client/pull/12/files#diff-e03ece5c80f80fe203c94520bdea5347